lint/util.go

```diff
@@ -0,0 +1,32 @@
+package lint
+
+import (
+	"go/ast"
+)
+
+func collectFuncDecls(f *ast.File) []*ast.FuncDecl {
+	var decls []*ast.FuncDecl
+	for _, decl := range f.Decls {
+		if decl, ok := decl.(*ast.FuncDecl); ok {
+			decls = append(decls, decl)
+		}
+	}
+	return decls
+}
+
+func collectFuncParams(decl *ast.FuncDecl) []*ast.Field {
+	var params []*ast.Field
```

Comment:
It seems line that this function will work only in param_name_checker. Do we really need array with recv, params and results in global case? Maybe we should move this function in param_name_checker or slpit it into 3 functions?

Reply:
Good point.
Fixed.